File: pages_reading.py
# https://www.spoj.com/problems/MWPZ031/



def pages_reading(pages,n):
    if '1' not in pages:
        return 0
    elif n == 1 and pages[0] == 1:
        return 1
    elif '1' in pages[0] and '1' not in pages[1:] and n > 1: 
        return 2
    elif ('0' in pages[0] and '0' not in pages[1:]) or '0' not in pages:
        return 1
    else:
        return 'NIGDY'


def main():
    data_sets = int(input())
    results = [0]*data_sets
    for i in range(data_sets):
        n = int(input())
        pages = input()
        results[i] = pages_reading(pages,n)
    for result in results: print(result)

main()












# def pages_reading(pages,n):
#     rounds = 0
#     if 1 not in pages: return rounds
#     else:
#         for i in range(n):
            
#             if pages[0] == 1 and i == 0:
                 
#                 for j in range(n):
#                     print('WArunek',pages[j])
#                     if pages[j] == 1: 
#                         pages[j] = 0
#                     else: 
#                         pages[j] = 1
#                 rounds += 1
#             else:
#                 pages.pop(0)
#                 for j in range(n-1):
#                     if j == 0 and pages[j] == 1 :
#                         pages[0] = 0                    
#                     elif j==0 and pages[j] == 0: 
#                         break

#                     if j != 0 and pages[j] == 1: 
#                         pages[j] = 0
#                     elif j != 0 and pages[j] == 0: 
#                         pages[j] = 1
#                 pages.append(0)
#                 rounds += 1
#             if 1 not in pages: 
#                 return rounds
        

#         if 1 in pages: 
#             return 'NIGDY'
#         else: 
#             return rounds


# def main():
#     data_sets = int(input())
#     results = [0]*data_sets
#     for i in range(data_sets):
#         n = int(input())
#         pages = list(map(int,input().split()))
#         results[i] = pages_reading(pages,n)
#     for result in results: print(result)

# main()

File: pages_reading.cpp
#include <iostream>
#include <vector>
using namespace std;

bool contains_one(const vector<int>& pages) {
    for (int page : pages) {
        if (page == 1)
            return true;
    }
    return false;
}

int pages_reading(vector<int>& pages, int n) {
    int rounds = 0;
    if (!contains_one(pages)) return rounds;

    for (int i = 0; i < n; ++i) {
        if (pages[0] == 1 && i == 0) {
            for (int j = 0; j < n; ++j) {
                if (pages[j] == 1)
                    pages[j] = 0;
                else
                    pages[j] = 1;
            }
            rounds += 1;
        }
        else {
            pages.erase(pages.begin());
            for (int j = 0; j < n - 1; ++j) {
                if (j == 0 && pages[j] == 1) {
                    pages[0] = 0;
                } else if (j == 0 && pages[j] == 0) {
                    break;
                }
                if (j != 0 && pages[j] == 1) {
                    pages[j] = 0;
                } else if (j != 0 && pages[j] == 0) {
                    pages[j] = 1;
                }
            }
            pages.push_back(0);
            rounds += 1;
            if (!contains_one(pages)) return rounds;
        }
    }

    if (contains_one(pages)) return -1;  // 'NIGDY'
    else return rounds;
}

int main() {
    int data_sets;
    cin >> data_sets;
    vector<int> results(data_sets);
    
    for (int i = 0; i < data_sets; ++i) {
        int n;
        cin >> n;
        vector<int> pages(n);
        for (int j = 0; j < n; ++j)
            cin >> pages[j];
        results[i] = pages_reading(pages, n);
    }
    
    for (int i = 0; i < data_sets; ++i) {
        if (results[i] == -1)
            cout << "NIGDY" << endl;
        else
            cout << results[i] << endl;
    }

    return 0;
}
